### PR TITLE
eclipse: modularized buildEclipse

### PR DESCRIPTION
**Pull request by muradu**
I modularized the Eclipse packages so that custom versions can easily be defined in, say, _~/.nixpkgs/config.nix_. I also added the latest version of the SDK (4.5.1) since that was my original requirement anyway.

I tried to do this in the most obviously equivalent way I could think of.

I **did not** change the _eclipse-platform_ version or even add the 4.5.1 platform release. Perhaps we should provide different versions like we do for the SDK. Something like:
-  rename _eclipse-platform_ to _eclipse_platform_45_
- add  _eclipse_platform_451_

But, I am unsure of the proper protocol and etiquette for renaming derivations, and I suspect that may be too much for this pull request.

### CODE REVIEW

**Comment by DamienCassou** [comment]
+1

If nobody disagrees, I will merge it.

File: pkgs/applications/editors/eclipse/buildEclipse.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{ stdenv, makeDesktopItem, freetype, fontconfig, libX11, libXrender, zlib, jre, glib, gtk, libXtst, webkitgtk2, makeWrapper, ... }:
```
Comment (edolstra) [inline pkgs/applications/editors/eclipse/buildEclipse.nix]: Don't use camelCase in filenames please.

**Comment by bjornfor** [comment]
+1

**Comment by rycee** [comment]
Yeah, this is nice. Change `buildEclipse.nix` to `build-eclipse.nix` or `eclipse-builder.nix`. If the point is to modularize, wouldn't it also make sense to make the `buildEclipse` variable an actual attribute in the `eclipses` set so that you could refer to it using `eclipses.buildEclipse`?

About `eclipse-platform` I've been meaning to add 4.5.1 and have something like
```nix
{
  eclipse-platform-4_5 = …
  eclipse-platform-4_5_1 = …
  eclipse-platform = eclipse-platform-4_5_1
}
```
that is `eclipse-platform` remains and will always point to the latest version of the platform. I've actually been running Eclipse Platform for quite some time and it's quite nice so really should go into Nixpkgs at some point.

**Comment by DamienCassou** [comment]
I'm closing this as @muradu does not seem to take care of the reviews. I opened another PR with the same content + the fixes for above feedback.

**Pull request closed without merging**